src/engine/enginemaster.cpp

```diff
@@ -443,22 +546,79 @@ void EngineMaster::process(const int iBufferSize) {
     }
 
     if (masterEnabled) {
-        // Mix the three channels together. We already mixed the busses together
-        // with the channel gains and overall master gain.
-        if (!m_pMasterTalkoverMix->toBool()) {
-            // Add Talkover to Master output
-            SampleUtil::copy4WithGain(m_pMaster,
-                    m_pOutputBusBuffers[EngineChannel::LEFT], 1.0,
-                    m_pOutputBusBuffers[EngineChannel::CENTER], 1.0,
-                    m_pOutputBusBuffers[EngineChannel::RIGHT], 1.0,
+        // Make master and booth mixes from the
+        // crossfader orientation buffers and talkover mix
+        SampleUtil::copy3WithGain(m_pMaster,
+            m_pOutputBusBuffers[EngineChannel::LEFT], 1.0,
+            m_pOutputBusBuffers[EngineChannel::CENTER], 1.0,
+            m_pOutputBusBuffers[EngineChannel::RIGHT], 1.0,
+            iBufferSize);
+
+        if (configuredTalkoverMixMode != TalkoverMixMode::DIRECT_MONITOR
+            && m_pNumMicsConfigured->get() > 0) {
+              m_pInputLatencyCompensationDelay->process(m_pMaster, iBufferSize);
+        }
+
+        if (boothEnabled) {
+            CSAMPLE boothGain = m_pBoothGain->get();
+
+            if (configuredTalkoverMixMode == TalkoverMixMode::MASTER) {
+                // Mix the talkover mix with the master output, but exclude it from
+                // the booth output.
+                if (m_bRampingGain) {
+                    SampleUtil::copy1WithRampingGain(m_pBooth, m_pMaster,
+                        m_boothGainOld, boothGain, iBufferSize);
+                } else {
+                    SampleUtil::copy1WithGain(m_pBooth, m_pMaster,
+                        boothGain, iBufferSize);
+                }
+
+                SampleUtil::copy2WithGain(m_pMaster,
+                    m_pMaster, 1.0,
+                    m_pTalkover, 1.0,
+                    iBufferSize);
+            } else if (configuredTalkoverMixMode == TalkoverMixMode::MASTER_AND_BOOTH) {
```

Comment:
That is not currently the case. The talkover mix is mixed into the master mix before the master mix is mixed with PFL for the headphone output. This PR does not change that.